[llvm-readobj][MachO] Add support for CPU_TYPE_ARM64_32

rjmansfield opened the pull request:
This adds support for ARM64_32 (ILP32) cpu type and subtype in llvm-readobj --file-headers. The support was previously added for llvm-objdump but was accidentally omitted from llvm-readobj.

rjmansfield committed:
[llvm-readobj][MachO] Add support for CPU_TYPE_ARM64_32

This adds support for ARM64_32 (ILP32) cpu type and subtype
in llvm-readobj --file-headers. The support was previously added for
llvm-objdump but was accidentally omitted from llvm-readobj.

jh7370 commented on llvm/tools/llvm-readobj/MachODumper.cpp:
```diff
@@ -111,15 +111,16 @@ const EnumEntry<uint32_t> MachOHeaderFileTypes[] = {
+    {"Arm64 (ILP32)", MachO::CPU_TYPE_ARM64_32},
+    {"Sparc", MachO::CPU_TYPE_SPARC},
+    {"PowerPC", MachO::CPU_TYPE_POWERPC},
+    {"PowerPC64", MachO::CPU_TYPE_POWERPC64},
```
I believe the intent of the original formatting is actually superior to the clang-formatted version. Please restore the style and add clang-format off/on comments around this block, so that the code format check no longer complains.

jh7370 commented on llvm/test/tools/llvm-readobj/MachO/file-headers.test:
```diff
@@ -158,3 +158,34 @@ FileHeader:
   sizeofcmds: 0
   flags:      0x00002000
+
+# RUN: yaml2obj %s --docnum=6 -o %t.arm64_32
```
There's a lot of duplication in this test, both in the YAML inputs and the FileCheck check prefixes. We can do better at least with the YAML version (I think it's probably not worth it for the FileCheck code due to the number of unique lines in some cases) by using the -D option to parameterise the input fields. (I say that, but I'm not 100% certain whether -D works for Mach-O yaml2obj, so you'd need to double-check). You can see plenty of examples in the ELF equivalents of these tests.

I wouldn't rework the whole test file (though if you want to, I'd support it in a different file), but you could do it for the ARM case above, combining it with this one.

jh7370 replied:
It looks to me like you've changed the FileCheck side, but not the YAML like I suggested. Changing the FileCheck bits is good, but the YAML was the bit I was more interested in. Could you explain why you haven't done that, please?

jh7370 reviewed (commented):
Looks basically fine. Just some test and formatting thoughts.

rjmansfield committed:
Address reviewer feedback.
 - Disable clang-format for MachOHeaderCpuTypes.
 - Refactor file-headers.test check patterns.

rjmansfield committed:
Consolidate yaml docs.

jh7370 reviewed (approved):
LGTM, thanks!
